src/bun.js/module_loader.zig

```diff
@@ -1929,22 +1942,18 @@ pub const ModuleLoader = struct {
                 }
 
                 // Pass along package.json type "module" if set.
-                const tag = brk: {
-                    if (parse_result.ast.exports_kind == .cjs and parse_result.source.path.isFile()) {
-                        const actual_package_json: *PackageJSON = package_json orelse brk2: {
-                            // this should already be cached virtually always so it's fine to do this
-                            const dir_info = (jsc_vm.transpiler.resolver.readDirInfo(parse_result.source.path.name.dirOrDot()) catch null) orelse
-                                break :brk .javascript;
-
-                            break :brk2 dir_info.package_json orelse dir_info.enclosing_package_json;
-                        } orelse break :brk .javascript;
-
-                        if (actual_package_json.module_type == .esm) {
-                            break :brk ResolvedSource.Tag.package_json_type_module;
-                        }
-                    }
-
-                    break :brk ResolvedSource.Tag.javascript;
+                const tag: ResolvedSource.Tag = switch (loader) {
+                    .json, .jsonc => .json_for_object_loader,
+                    .js, .jsx, .ts, .tsx => if (parse_result.source.path.isFile()) brk: {
```

Comment:
The `isFile` isn't necessary 
```suggestion
                    .js, .jsx, .ts, .tsx => brk: {
```